core/gallery/gallery.go

```diff
@@ -0,0 +1,133 @@
+package gallery
+
+import (
+	_ "embed"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"regexp"
+	"strings"
+	"time"
+
+	strip "github.com/grokify/html-strip-tags-go"
+	"github.com/xeipuuv/gojsonschema"
+	"golang.org/x/net/html"
+
+	"github.com/anyproto/anytype-heart/pb"
+	"github.com/anyproto/anytype-heart/util/uri"
+)
+
+const timeout = time.Second * 30
+
+type schemaResponse struct {
+	Schema string `json:"$schema"`
+}
+
+// whitelist maps allowed hosts to regular expressions of URL paths
+var whitelist = map[string]*regexp.Regexp{
+	"raw.githubusercontent.com": regexp.MustCompile(`/anyproto.*`),
```

Comment:
this matches regexp as well:
```
/foo/bar/anyproto
```

maybe we need to add `^` at the beginning of regex?